apis/Google.Cloud.Diagnostics.AspNet/Google.Cloud.Diagnostics.AspNet/ErrorReporting/ErrorReportingExceptionFilter.cs

```diff
@@ -0,0 +1,150 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using Google.Cloud.ErrorReporting.V1Beta1;
+using System.Threading.Tasks;
+using System.Web;
+using System.Web.Mvc;
+
+namespace Google.Cloud.Diagnostics.AspNet
+{
+    /// <summary>
+    ///  Google Cloud Error Reporting <see cref="IExceptionFilter"/>.
+    /// </summary>
+    /// 
+    /// <example>
+    /// <code>
+    /// public static void RegisterGlobalFilters(GlobalFilterCollection filters)
+    /// {
+    ///   // Add a catch all for the uncaught exceptions.
+    ///   string projectId = "[Google Cloud Platform project ID]";
+    ///   string serviceName = "[Name of service]";
+    ///   string version = "[Version of service]";
+    ///   // Add a catch all for the uncaught exceptions.
+    ///   filters.Add(ErrorReportingExceptionFilter.Create(projectId, serviceName, version));
+    /// }
+    /// </code>
+    /// </example>
+    /// 
+    /// <remarks>
+    /// Reports unhandled exceptions to Google Cloud Error Reporting.
+    /// The filter should be registered first in ASP.NET MVC versions 1 and 2 and last
+    /// in ASP.NET MVC versions 3 and higher. See:
+    /// https://msdn.microsoft.com/en-us/library/system.web.mvc.iexceptionfilter.onexception(v=vs.118).aspx
+    /// Docs: https://cloud.google.com/error-reporting/docs/
+    /// </remarks>
+    public class ErrorReportingExceptionFilter : IExceptionFilter
+    {
+        // The Google Cloud Platform project id as a resource name.
+        private readonly ProjectName _projectResourceName;
+
+        // The service context in which this error has occurred.
+        // See: https://cloud.google.com/error-reporting/reference/rest/v1beta1/projects.events#ServiceContext
+        private readonly ServiceContext _serviceContext;
+
+        private readonly Task<ReportErrorsServiceClient> _clientTask;
+
+        /// <summary>
+        /// Creates an instance of <see cref="ErrorReportingExceptionFilter"/>
+        /// </summary>
+        /// <param name="clientTask">The Error Reporting client.</param>
+        /// <param name="projectId">The Google Cloud Platform project ID.</param>
+        /// <param name="serviceName">An identifier of the service, such as the name of the executable or job.</param>
+        /// <param name="version">Represents the source code version that the developer provided.</param> 
+        public static ErrorReportingExceptionFilter Create(
+            ReportErrorsServiceClient client, string projectId, string serviceName, string version) =>
+                new ErrorReportingExceptionFilter(Task.FromResult(client), projectId, serviceName, version);
+
+        /// <summary>
+        /// Creates an instance of <see cref="ErrorReportingExceptionFilter"/>
+        /// </summary>
+        /// <param name="clientTask">The Error Reporting client.</param>
```

Comment:
This should probably have a slightly different description to indicate that it is a task. Perhaps something like "A task which will produce the Error Reporting client."